src/main/java/gov/nasa/cumulus/CNMResponse.java

```diff
@@ -58,19 +58,30 @@ public void handleRequestStreams(InputStream inputStream, OutputStream outputStr
 	public static JsonObject getResponseObject(String exception) {
 		JsonObject response = new JsonObject();
 
-		if(exception == null || new String("").equals(exception) ||  new String("None").equals(exception) ||  new String("\"None\"").equals(exception)){
-			//success
+        if (exception == null || exception.isEmpty() || exception.equals("None") || exception.equals("\"None\"")) {
+            //success
 			AdapterLogger.LogInfo(CNMResponse.class.getName() + " status: SUCCESS");
 			response.addProperty("status", "SUCCESS");
-		}else{
+		} else {
 			//fail
 			AdapterLogger.LogWarning(CNMResponse.class.getName() + " status: FAILURE");
 			response.addProperty("status", "FAILURE");
 
 			//logic for failure types here
-			JsonObject workflowException = new JsonParser().parse(exception).getAsJsonObject();
+            String error;
+            String causeString;
+            JsonObject workflowException;
+
+            // PODAAC-2552 - handle general exceptions being thrown
+            if (exception.equalsIgnoreCase("CNMResponse Exception")) {
+                error = exception;
+                causeString = "Unknown cause, CNMResponse likely threw an exception";
+            } else {
+                workflowException = new JsonParser().parse(exception).getAsJsonObject();
+                error = workflowException.get("Error").getAsString();
+                causeString = workflowException.get("Cause").getAsString();
+            }
```

Comment:
Can you please reformat the code?   The original one was all indented correctly.  You can easily do that if you are using intelliJ IDE